src/editor/actions/file-request-actions.js

```diff
@@ -0,0 +1,169 @@
+import {
+  loadFileFromServer,
+  saveFileToServer,
+  deleteFileOnServer
+} from "../../shared/utils/file-operations";
+import {
+  getNotebookID,
+  fileExists,
+  getFileID,
+  validateFileExistence,
+  validateFileAbsence,
+  validateFetchType
+} from "../tools/server-tools";
+
+function messagePasserIsFunction(passer) {
+  return typeof passer === "function";
+}
+
+function validateMessagePasser(passer) {
+  if (!messagePasserIsFunction(passer)) {
+    throw new Error("message passer must be a function");
+  }
+}
+
+export function addFileToNotebook(filename, lastUpdated, fileID) {
+  return {
+    type: "ADD_FILE_TO_NOTEBOOK",
+    filename,
+    lastUpdated,
+    fileID
+  };
+}
+
+export function deleteFileFromNotebook(fileID) {
+  return {
+    type: "DELETE_FILE_FROM_NOTEBOOK",
+    fileID
+  };
+}
+
+function onFileOperationSuccess(fileRequestID, messagePasser, fileOrResponse) {
```

Comment:
ah yeah, I think overall I was originally in the position of creating a cyclical reference in this part of the code base, which is why I went this direction (vs the default arg). Will still need to check if that is still true!

Reply:
it won't create a circular ref if you import `messagePasserEditor` -- the message passer was originally implemented to let us avoid such circularities

Reply:
hmm ... ok. This definitely wasn't true during one part of this journey, but it might not have been from the message passer. All the same will go with the default arg

Reply:
oh, I see why the circular ref was there like 20 commits back and wouldn't be now – thanks for catching